src/libsync/clientsideencryption.cpp

```diff
@@ -1260,6 +1335,38 @@ void ClientSideEncryption::getPublicKeyFromServer(const AccountPtr &account)
     job->start();
 }
 
+void ClientSideEncryption::getServerPublicKeyFromServer(const AccountPtr &account)
+{
+    qCInfo(lcCse()) << "Retrieving public key from server";
+    auto job = new JsonApiJob(account, baseUrl() + "server-key", this);
+    connect(job, &JsonApiJob::jsonReceived, [this, account](const QJsonDocument& doc, int retCode) {
+            if (retCode == 200) {
+                auto serverPublicKey = doc.object()["ocs"].toObject()["data"].toObject()["public-key"].toString();
```

Comment:
@mgallien I am wondering, won't it be better to use `toByteArray()` conversion and then pass `QByteArray` into `checkServerPublicKeyValidity` and change its argument to accept `const QByteArray &` instead of `const QString &` since we are converting it inside anyway? Or, you could still use `toString` but also convert the string with `toLatin1` here. This way, you would avoid conversion inside the function and it will be more clear that it works with `QByteArra`. Just a suggestion.

Also, `const auto serverPublicKey`?